swig/include/java/gdal_java.i

```diff
@@ -304,12 +305,14 @@ static CPLErr DatasetRasterIO( GDALDatasetH hDS, GDALRWFlag eRWFlag,
 %apply (char *regularArrayOut, long nRegularArraySizeOut) { (char *regularArrayOut, long nRegularArraySizeOut) };
 %apply (short *regularArrayOut, long nRegularArraySizeOut) { (short *regularArrayOut, long nRegularArraySizeOut) };
 %apply (int *regularArrayOut, long nRegularArraySizeOut) { (int *regularArrayOut, long nRegularArraySizeOut) };
+%apply (long *regularArrayOut, long nRegularArraySizeOut) { (long *regularArrayOut, long nRegularArraySizeOut) };
```

Comment:
- The C type long is only 64bit large on non-Windows 64-bit platforms. A more portable type would be int64_t
Same remark for all other places you edited
- swig/include/java/typemaps_java.i should be modified to define the 2 new typemaps you've declared. See existing typemaps ``(int *regularArrayOut, long nRegularArraySizeOut)`` and  ``(int *regularArrayIn long nRegularArraySizeIn)``
- we'd definitely need some testing. For example in swig/java/apps/GDALTestIO.java, add a new method to write 64-bit integers in a GDT_Int64 MEM dataset and read them back